### PR TITLE
Allow hostnames to contain upper-case letters

### PR DESCRIPTION
**Pull request by arsing**
This change removes the additional restriction that hostnames must not contain
upper-case A-Z. The reason for the restriction was to be consistent with
Kubernetes services, but we've received feedback from non-Kubernetes users
that they'd like to use upper-case.

### CODE REVIEW

File: edgelet/iotedge/src/check/mod.rs [hunk 1]
```diff
@@ -1964,22 +1964,26 @@ fn is_rfc_1035_valid(name: &str) -> bool {
             Some(c) => c,
             None => return false,
         };
-        if first_char < 'a' || first_char > 'z' {
+        let first_char_invalid =
+            (first_char < 'a' || first_char > 'z') && (first_char < 'A' || first_char > 'Z');
```
Comment (darobs) [inline edgelet/iotedge/src/check/mod.rs]: Could this be replaced by  `is_ascii_alphabetic()`?

Comment (arsing) [reply edgelet/iotedge/src/check/mod.rs]: Oh yeah, good point. Done.

File: edgelet/iotedge/src/check/mod.rs [hunk 2]
```diff
@@ -1964,22 +1964,26 @@ fn is_rfc_1035_valid(name: &str) -> bool {
...
+        if first_char_invalid {
             return false;
         }
 
-        if label
-            .chars()
-            .any(|c| (c < 'a' || c > 'z') && (c < '0' || c > '9') && c != '-')
-        {
+        let any_char_invalid =
+            |c| (c < 'a' || c > 'z') && (c < 'A' || c > 'Z') && (c < '0' || c > '9') && c != '-';
```
[the remark this answers is not in the document]
Comment (arsing) [reply edgelet/iotedge/src/check/mod.rs]: Done.

**Comment by arsing** [comment]
Done.

**Pull request merged**